### ISSUE TITLE
save button does not go away

### ISSUE THREAD

**Issue by velociwabbit** [marmelab/admin-on-rest#376]
Most likely I have redirect problems with redux-forms or with crudUpdate / crudSaga but this part of the code is a bit complex for my limited abilities.

I finally was able to match my rest database components with the jsonServer "rest"calls and have confirmed that the correct data is being updated on the server but somehow the flag for being finished is not being set before Redux-Form  takes over and Redux-form keeps spinning waiting for that flag to be set.

My question where and how isthat  **_save is finished go back to the previous component flag_**  set?

This is also related to another problem I am having when a new feature (being able to limit table columns dynamically.) The component works but in order to do so I have to re-select from the left side menu lists. What i would really  like to do is trigger a redraw of the entire component without any other user interaction.  Is this a crudRoute issue or somewhere else?

Any help would be appreciated this redux-form / saga stuff really spins me in circles.

**Comment by velociwabbit** [comment]
Accidental close.

I am thinking that redux should be renamed rubix as in the cube!.  Very small changes have very big consequences.

**Comment by fzaninotto** [comment]
Save goes back to the previous state in [`crudResponse`](https://github.com/marmelab/admin-on-rest/blob/master/src/sideEffect/saga/crudResponse.js). This happens only if no exception is thrown by your `restClient` (cf https://github.com/marmelab/admin-on-rest/blob/master/src/sideEffect/saga/crudFetch.js#L21-L26).

**Comment by velociwabbit** [comment]
thanks

I noticed that you changed crudfetch between release 7 and release 8 to
remove the successSideEffects and failureSideEffects and replaced them with
a crudResponse.

in order to figure out my last bug I ended up consolidating 7.2 version of
success, failure, referenceFetch and index.js into one method but I cannot
seem to get the newer version with meta carrying the  response routing to
work.

My question is whether this change was brought about by a bug in 7.x or are
there other parts of the code that were also affected by it in order to
make it work?



On Tue, Mar 7, 2017 at 3:06 AM, Francois Zaninotto <notifications@github.com
